third_party/java/arcs/build_defs/internal/schemas.bzl

```diff
@@ -164,16 +159,110 @@ def arcs_kt_schema(
         )
     return {"outs": outs, "deps": outdeps}
 
+def _arcs_ts_preproc_impl(ctx):
+    outputs = [
+        ctx.actions.declare_file(x.basename, sibling = x)
+        for x in ctx.files.srcs
+    ]
+
+    ctx.actions.run_shell(
+        inputs = ctx.files.srcs,
+        outputs = outputs,
+        arguments = [src.path for src in ctx.files.srcs],
+        command = """
+        for i in "$@"; do
+          out="$(pwd)/bazel-out/host/bin/$i"
+          sed -e 's/-web.js/-node.js/g' $i > "$out"
+        done
+        """,
+    )
+
+    return DefaultInfo(files = depset(outputs))
+
+arcs_ts_preprocessing = rule(
+    implementation = _arcs_ts_preproc_impl,
+    attrs = {
+        "srcs": attr.label_list(allow_files = [".ts", ".d.ts"]),
+    },
+    doc = """Converts to node platforms for the arcs ts runtime.""",
```

Comment:
A more elaborate documentation would be useful.

Maybe something along the lines of:

```
Processes Arcs TypeScript sources to refer to node variants of the dependencies in the platform/ directory.
```